### ISSUE TITLE
Elucidating the Design Space of Diffusion-Based Generative Models

### ISSUE THREAD

**Issue by reyllama** [reyllama/paper-reviews#192]
# TL; DR

- Take each components of diffusion models, e.g., sampling, noise schedule, network architecture and preconditioning and both theoretically and empirically analyze their design choices.

# Motivation

- *The theory and practice of diffusion-based generative models are currently unnecessarily convoluted*, resulting in *a tightly coupled package where no individual component can be modified without breaking the entire system*.

# Formulation

### ODE Solver

- 2nd order Runge-Kutta family solver (Heun's solver) offers better trade-off than commonly used Euler method.

### Generative Sampling and Noise Schedule

- People commonly believe that the training and the sampling process are in some way coupled, but that turns out to be false. From the viewpoint of generative sampling, the trained denoiser is a simple black box.

- The step size should decrease monotonically but doesn't have to be sample-dependent.

![image](https://user-images.githubusercontent.com/46308407/203732396-e3244e1c-6815-473f-8a47-d6719acaef60.png)

- It is beneficial to set noise schedule so that the high-curvature region in the ODE trajectory is minimized to obtain lower truncation errors.

![image](https://user-images.githubusercontent.com/46308407/203732634-d192d87a-66bc-4f96-9822-5954da657b56.png)

### Extension to Stochastic Sampling

- Although deterministic sampling presents advantages such as invertibility of an arbitrary real image, stochastic sampling generally yields better quality outputs.
- Stochastic sampling process structurally resemble Langevin dynamics, and can be thought of having the benefit of being able to make up for past errors (in a sequence of denoising steps).

![image](https://user-images.githubusercontent.com/46308407/203733056-cb18abc3-c483-45a9-b409-ce6cbfad0899.png)

![image](https://user-images.githubusercontent.com/46308407/203733214-7b51211e-3036-49bd-a1a7-d6621dfc37cd.png)

- The authors introduce a novel stochastic sampler that makes explicit Langevin-like "churn" of adding and removing noise.
- It resembles predictor-corrector sampler of Song et. al (ICLR21).

![image](https://user-images.githubusercontent.com/46308407/203734329-83b0a40b-e0b4-4ca2-8ccc-0a79af97c90f.png)

### Preconditioning and Training

- Most DMs opt to predict the noise instead of the denoised sample itself.
- It is certainly beneficial in low-variance regime (later sampling stage), but can result in amplifying network errors especially in the high-variance stage.
- Hence, the authors present a new formulation where the model predicts denoised sample at early stages and the noise at later stages.

![image](https://user-images.githubusercontent.com/46308407/203733954-553609ee-690f-41d6-9c84-c6e650b64632.png)
![image](https://user-images.githubusercontent.com/46308407/203733986-501bec88-a792-4537-bdb5-e22978e30f02.png)

- Also, sampling noise from a log-normal distribution renders the training more efficient, by sampling more on the relevant regime (in the middle).
- Generative augmentations such as non-leaking augmentations help diffusion models as well.

![image](https://user-images.githubusercontent.com/46308407/203734264-62eff532-4b57-41f3-a85e-fcbfcdfb98af.png)

# Inspiration

- The authors have once again proven themselves masters of architecture designs.
- Some insights are very intuitive (e.g., noise sampling, preconditioning) but others are more difficult to understand.
- Definitely need to revisit score-based generative modelings and neural ODEs.
